src/LibSodium/Bindings/Utils.hs

```diff
@@ -0,0 +1,152 @@
+{-# LANGUAGE CApiFFI #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE Trustworthy #-}
+{-# OPTIONS_GHC -Wno-unused-imports #-}
+
+-- | Module: LibSodium.Bindings.Utils
+-- Description: Helpers defined in utils.h
+-- Copyright: (C) Seth Livy 2022
+-- License: BSD-3-Clause
+-- Maintainer: sethlivy@gmail.com
+-- Stability: Stable
+-- Portability: GHC only
+
+module LibSodium.Bindings.Utils
+  ( -- * Low-level binding
+    sodiumMemcmp
+  , sodiumBin2Hex
+  , sodiumHex2Bin
+  , sodiumBin2Base64
+  , sodiumBase642Bin
+
+    -- * Constants
+  , sodiumBase64VariantOriginal
+  , sodiumBase64VariantOriginalNoPadding
+  , sodiumBase64VariantURLSafe
+  , sodiumBase64VariantURLSafeNoPadding
+  )
+where
+
+import Foreign (Ptr)
+import Foreign.C (CChar (CChar), CInt (CInt), CSize (CSize), CUChar)
+import Foreign.C.String
+
+-- | Constant-time comparison function.
+--
+-- This function is not a lexicographic comparator and should be never
+-- used for this purpose. It should only be used when comparing two pieces
+-- of secret data, such as keys or authentication tags.
+foreign import capi "sodium.h sodium_memcmp"
+  sodiumMemcmp
+    :: Ptr ()
+    -- ^ First pointer to some secret data.
+    -> Ptr ()
```

Comment:
`()` is _not_ the same as C's `void`. If anything, this should probably be `Ptr CUChar`.

Reply:
I've changed both of these arguments to `Ptr CUChar`.